### PR TITLE
Replace the System.exit calls with getActivity.finishAffinity()

### LINKED ISSUE #6428: Replace the System.exit calls with getActivity.finishAffinity()
### Checklist

- [x] I checked, but didn't find any duplicates (open OR closed) of this issue in the repo. 
- [x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [x] This issue contains only one feature request. I will open one issue for every feature I want to request.

#### Describe the feature you want
From https://github.com/TeamNewPipe/NewPipe/pull/6402#issuecomment-852970989
> could we take the opportunity and replace the System.exit calls with getActivity.finishAffinity() and launch of the main activity?
>so
>System.exit(0);
>to
>`getActivity().finishAffinity();
>startActivity(getActivity().getPackageManager().getLaunchIntentForPackage(getActivity().getPackageName()));

#### Is your feature request related to a problem? Please describe it

 I don't know, @Redirion Whats the upside if this?

#### Additional context

- Replace here https://github.com/TeamNewPipe/NewPipe/blob/332b90d6c18b52bc671271ec7589c072b71ff785/app/src/main/java/org/schabi/newpipe/settings/ContentSettingsFragment.java#L273

#### Testcriteria
(for whoever picks this up)
- Settings are still loaded correctly after importing them

### PR DESCRIPTION
**Pull request by danielmbutler**
Replaced System.exit call with finishAffinity() and start MainActivity call, tested by changing some settings values, exporting my configuration, reversing the changes, and then reimporting the settings, once the MainActivity was restarted I could then see my chosen settings imported.

Also updated ExitActivity.java to remove System.Exit call and replace with start MainActivity

#### What is it?
- [ ] Bugfix (user facing)
- [ ] Feature (user facing)
- [X] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR
Replaced System.Exit calls with  getActivity.finishAffinity() in the below 2 files

> NewPipe/app/src/main/java/org/schabi/newpipe/settings/ContentSettingsFragment.java
> src/main/java/org/schabi/newpipe/ExitActivity.java

#### Fixes the following issue(s)

- Fixes #6428 

#### APK testing 
I ran the APK on an emulator and went through the settings flow and ensured that my changes persisted after the importing task had finished and the app was restarted.

#### Due diligence
- [X] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mhmdanas** [comment]
Thanks for making a PR! Please avoid putting the issue number in the PR title.

**Comment by mhmdanas** [comment]
Note that the "APK testing" section is for putting a link to the APK (from the "Checks" tab of the PR; the APK isn't there currently because you're a first-time contributor so a maintainer has to manually approve the checks to be run), not for how you tested it yourself.

**Comment by danielmbutler** [comment]
Apologies for "Apk Testing" section I will remember this in future :)

File: app/src/main/java/org/schabi/newpipe/ExitActivity.java
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/ExitActivity.java:53]: Create a common `restartApp(final Activity activity)` method for this in the `util/NavigationHelper` class and reuse it in both places

**Review by Stypox — CHANGES_REQUESTED** [review]
Thank you!

File: app/src/main/java/org/schabi/newpipe/ExitActivity.java
Comment (danielmbutler) [reply app/src/main/java/org/schabi/newpipe/ExitActivity.java]: I've created the common method and updated ExitActivity.java and ContentSettingsFragment.java :), changes have been pushed to my branch

File: app/src/main/java/org/schabi/newpipe/util/NavigationHelper.java
Comment (mhmdanas) [inline app/src/main/java/org/schabi/newpipe/util/NavigationHelper.java:607]:
```suggestion
     * Finish provided <code>Activity</code> and then start <code>MainActivity</code>.
     *
     * @param activity the activity to finish
     * @param context the context to use when creating the <code>Intent</code>
```

Haven't checked this; should wait for CI.

**Review by mhmdanas — CHANGES_REQUESTED** [review]
LGTM, just a bunch of minor nits.

File: app/src/main/java/org/schabi/newpipe/util/NavigationHelper.java
Comment (Redirion) [inline app/src/main/java/org/schabi/newpipe/util/NavigationHelper.java:604]: not just the provided activity is finished, all activities sharing the same task. As NewPipe has the Manifest flag "singleTask", all activies are finished. SingleTask is set here: https://github.com/TeamNewPipe/NewPipe/blob/d6e0bd8c26af1b7818e025d8dcda2cb69f636fab/app/src/main/AndroidManifest.xml#L31

https://developer.android.com/reference/android/app/Activity#finishAffinity()
"Finish this activity as well as all activities immediately below it in the current task that have the same affinity."

Comment (danielmbutler) [reply app/src/main/java/org/schabi/newpipe/util/NavigationHelper.java:604]: Okay, would the following verbiage work?

"Finish this activity as well as all activities running below it and then start MainActivity".

Comment (Redirion) [reply app/src/main/java/org/schabi/newpipe/util/NavigationHelper.java:604]: sounds good to me

**Comment by Redirion** [comment]
@XiangRongLin I think we can just squash the commits when we merge this and add our own summary, so no worries there

**Comment by XiangRongLin** [comment]
That would need to be done by him, because otherwise i think it counts the contribution towards the one squashing and merging it and not the PR opener

**Review by Redirion — DISMISSED** [review]
lgtm

**Comment by danielmbutler** [comment]
let me know if all is good then I will squash the commit messages.

**Comment by Redirion** [comment]
yes, all good, go ahead please

**Comment by danielmbutler** [comment]
Should be sorted now

**Comment by XiangRongLin** [comment]
@Stypox @mhmdanas You two have to approve it, because you requested changes

**Comment by Redirion** [comment]
I still see 9 commits. Looks like the squashing did not work. Can you try again @danielmbutler ?

**Review by mhmdanas — APPROVED** [review]

**Comment by danielmbutler** [comment]
Hi Guys I've been trying to squash my previous commits but it seems to merge them all together and then commit the merge on top of previous commits, not sure what I'm doing wrong

**Comment by danielmbutler** [comment]
After multiple tries, I figured it out lol

**Review by Stypox — APPROVED** [review]
Perfect, thank you!

**Pull request merged**
